src/peerstate.rs

```diff
@@ -248,14 +235,10 @@ impl Peerstate {
             let old_public_fingerprint = self.public_key_fingerprint.take();
             self.public_key_fingerprint = Some(public_key.fingerprint());
 
-            if old_public_fingerprint.is_none()
-                || self.public_key_fingerprint.is_none()
-                || old_public_fingerprint != self.public_key_fingerprint
+            if old_public_fingerprint.is_some()
+                && old_public_fingerprint != self.public_key_fingerprint
             {
-                self.to_save = Some(ToSave::All);
-                if old_public_fingerprint.is_some() {
-                    self.fingerprint_changed = true;
-                }
+                self.fingerprint_changed = true;
```

Comment:
Nice that we got rid of this code! UNIQUE contraints are awesome

 (I added the old, convoluted code since it was the only w/out knowing about UNIQUE contraints)